pkg/exchange/bitget/stream.go

```diff
@@ -116,6 +137,36 @@ func (s *Stream) handleBookEvent(o BookEvent) {
 	}
 }
 
+// ping implements the bitget text message of WebSocket PingPong.
+func (s *Stream) ping(ctx context.Context, conn *websocket.Conn, cancelFunc context.CancelFunc) {
+	defer func() {
+		log.Debug("ping worker stopped")
+		cancelFunc()
+	}()
+
+	var pingTicker = time.NewTicker(pingInterval)
+	defer pingTicker.Stop()
+
+	for {
+		select {
+
+		case <-ctx.Done():
+			return
+
+		case <-s.CloseC:
+			return
+
+		case <-pingTicker.C:
+			err := conn.WriteMessage(websocket.TextMessage, pingBytes)
```

Comment:
when you need to write message, you need to lock it, or the built-in ping pong handler could cause panic